riklet/src/net_utils.rs

```diff
@@ -0,0 +1,82 @@
+use std::net::Ipv4Addr;
+
+use futures_util::TryStreamExt;
+use rand::Rng;
+use rtnetlink::new_connection;
+use tracing::{trace, warn};
+use utils::net::mac::MacAddr;
+
+#[tracing::instrument()]
+pub async fn set_link_up(iface_name: String) -> Result<(), rtnetlink::Error> {
+    trace!("link {} up", &iface_name);
+    let (connection, handle, _) = new_connection().unwrap();
```

Comment:
🟠 I don't think this unwrap is safe

Reply:
It's fairly safe, as it won't be able to unwrap only if `rtnetlink` is not available as a socket. That said, linux kernel uses rtnetlink for routing ([docs](https://man7.org/linux/man-pages/man7/rtnetlink.7.html)). If the user doesn't have rtnetlink, it means he doesn't have Linux, and we don't support MacOS / Linux